File: ABC/351-400/393/b.py
def main():
    s = list(input())
    n = len(s)
    ans = 0
    for d in range(1,n):
        for i in range(n):
            if s[i] != 'A':
                continue
            if i+2*d >= n:
                break
            if s[i]+s[i+d]+s[i+2*d] == "ABC":
                ans += 1
    print(ans)
    
if __name__ == "__main__":
    main()

File: ABC/351-400/393/b.cpp
#include <bits/stdc++.h>
using namespace std;
#define rep(i,n) for(int i=0;i<n;++i)

int main(){
    string s;
    cin >> s;
    int n = s.size();
    int ans = 0;
    for(int d=1;d<n;d++){
        rep(i,n){
            if(s[i] != 'A') continue;
            if(i+2*d >= n) continue;
            string now =  "";
            now += s[i];
            now += s[i+d];
            now += s[i+2*d];
            if(now == "ABC") ans++;
        }
    }
    cout << ans << endl;
    return 0;
}